.github/workflows/run_shark_tank.yml

```diff
@@ -0,0 +1,140 @@
+name: SHARK
+
+on:
+  # schedule:
+  #  - cron: '0 10 * * *'
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  setup:
+    runs-on: ubuntu-20.04
+    env:
+      # The commit being checked out is the merge commit for the PR. Its first
+      # parent will be the tip of main.
+      BASE_REF: HEAD^
+      PR_DESCRIPTION: ${{ github.event.pull_request.body }}
+    outputs:
+      should-run: ${{ steps.configure.outputs.should-run }}
+      ci-stage: ${{ steps.configure.outputs.ci-stage }}
+      runner-env: ${{ steps.configure.outputs.runner-env }}
+      runner-group: ${{ steps.configure.outputs.runner-group }}
+      write-caches: ${{ steps.configure.outputs.write-caches }}
+      shark-sha: ${{ steps.shark.outputs.shark-sha }}
+      artifact-upload-dir: ${{ steps.shark.outputs.artifact-upload-dir }}
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2
+        with:
+          # We need the parent commit to do a diff
+          fetch-depth: 2
+      - name: "Configuring CI options"
+        id: configure
+        run: |
+          # Just informative logging. There should only be two commits in the
+          # history here, but limiting the depth helps when copying from a local
+          # repo instead of using checkout, e.g. with
+          # https://github.com/nektos/act where there will be more.
+          git log --oneline --graph --max-count=3
```

Comment:
Not sure if I understand this part. It seems like we fetch 2 commits just to show their logs here? Since this workflow will run on the main branch, why don't we just look at the main branch log? (when debugging)

Reply:
Oh, I saw this is also on the main workflow. Feel free to keep it as it is.